### PR TITLE
Add mParticle adapter

### PR DESCRIPTION
**Pull request by rmi22186**
This PR adds an adapter for mParticle.

https://docs.mparticle.com
https://www.mparticle.com

Please let me know if you have any questions.

### CODE REVIEW

**Comment by rmi22186** [comment]
Hi @jherdman! I saw that you most recently approved a PR for ember-metrics, so thought I'd ping you to take a look at this one.  Let me know if you have any questions!

File: addon/metrics-adapters/m-particle.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    const config = assign({}, this.config);
+    const { apiKey, options } = config;
+    assert(
+      `[ember-metrics] You must pass a valid \`apiKey\` to the ${this.toString()} adapter`,
+      apiKey
+    );
+
+    window.mParticle = {
+      config: {
+        isDevelopmentMode: true, //switch to false (or remove) for production
```
Comment (jherdman) [inline addon/metrics-adapters/m-particle.js:23]: Will this cause issues for people using this adapter?

File: addon/metrics-adapters/m-particle.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+  init() {
+    const config = assign({}, this.config);
+    const { apiKey, options } = config;
+    assert(
+      `[ember-metrics] You must pass a valid \`apiKey\` to the ${this.toString()} adapter`,
+      apiKey
+    );
+
+    window.mParticle = {
```
Comment (jherdman) [inline addon/metrics-adapters/m-particle.js:21]: Is everything from here on down provided by mParticle?

**Review by jherdman — COMMENTED** [review]
Sorry for not getting to this sooner. A few initial questions for you.

File: addon/metrics-adapters/m-particle.js (hunk 1 above)
Comment (rmi22186) [reply addon/metrics-adapters/m-particle.js:23]: good catch!  i'll get this from the config value passed from the environment.js file

File: addon/metrics-adapters/m-particle.js (hunk 2 above)
Comment (rmi22186) [reply addon/metrics-adapters/m-particle.js:21]: that's right, it's all part of our snippet to get mParticle onto the page.  I can add that as a comment

File: addon/metrics-adapters/m-particle.js (hunk 1 above)
Comment (jherdman) [reply addon/metrics-adapters/m-particle.js:23]: @rmi22186 just wondering if you had a bit of time to finish off this PR

**Comment by jherdman** [comment]
Ping @rmi22186

**Comment by jherdman** [comment]
Closing due to inactivity. Please feel free to reopen if you're interested in picking up this work again.

**Pull request closed without merging**